packages/samples/mvp-dapp/src/app.controller.ts

```diff
@@ -1,95 +1,124 @@
 import { KuaiRouter } from '@ckb-js/kuai-io'
-import { helpers } from '@ckb-lumos/lumos'
-import { Actor } from '@ckb-js/kuai-models'
-import type { types } from '@ckb-js/kuai-io'
+import { HashType, HexString, Script } from '@ckb-lumos/lumos'
+import { ActorReference, ProviderKey, UpdateStorageValue } from '@ckb-js/kuai-models'
 import { appRegistry } from './actors'
-import { Tx } from './views/tx.view'
 import { Load } from './views/load.view'
 import { Read } from './views/read.view'
-
-const { TransactionSkeleton } = helpers
+import { OmnilockModel } from './omnilock/omnilock.model'
+import { computeScriptHash } from '@ckb-lumos/base/lib/utils'
+import resourceBinding from './resource-binding'
+import { RecordModel, StoreType } from './record/record.model'
+import { DAPP_DATA_PREFIX } from './const'
 
 const router = new KuaiRouter()
-
-router.post('/claim', async (ctx) => {
-  const txSkeleton = new TransactionSkeleton()
-  // findCells
-
-  ctx.ok(Tx.toJsonString(txSkeleton, [], []))
+function createCellPattern(lock: Script) {
+  return (value: UpdateStorageValue) => {
+    const cellLock = value.cell.cellOutput.lock
+    return cellLock.args === lock?.args && cellLock.codeHash === lock?.codeHash && cellLock.hashType === lock?.hashType
+  }
+}
+
+function createRecordPattern(lock: Script) {
+  return (value: UpdateStorageValue) => {
+    const cellLock = value.cell.cellOutput.lock
+    return (
+      cellLock.args === lock?.args &&
+      cellLock.codeHash === lock?.codeHash &&
+      cellLock.hashType === lock?.hashType &&
+      value.cell.data.startsWith(DAPP_DATA_PREFIX)
+    )
+  }
+}
+
+function getOmnilockModel(lock: Script): OmnilockModel {
+  const lockHash = computeScriptHash(lock)
+  const actorRef = new ActorReference('omnilock', `/${lockHash}/`)
+  let omnilockModel = appRegistry.find<OmnilockModel>(actorRef.uri)
+  if (!omnilockModel) {
+    class NewStore extends OmnilockModel {}
+    Reflect.defineMetadata(ProviderKey.Actor, { ref: actorRef }, NewStore)
+    Reflect.defineMetadata(ProviderKey.CellPattern, createCellPattern(lock), NewStore)
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    appRegistry.bind(NewStore as any)
+    omnilockModel = appRegistry.find<OmnilockModel>(actorRef.uri)
+    if (!omnilockModel) throw new Error('ominilock bind error')
+    resourceBinding.register(lock, undefined, actorRef.uri, lockHash)
```

Comment:
I think maybe it is better to interact with resource binding manager by message bus.